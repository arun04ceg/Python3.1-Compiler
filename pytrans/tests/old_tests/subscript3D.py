a = [[[1, 2, 3], [3, 4, 5]]]
a[0][0][0] = 2
print(a[0][0][0])

