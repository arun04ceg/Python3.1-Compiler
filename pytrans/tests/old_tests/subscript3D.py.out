(program
 (define a (void))
 (set-global! a (py-list* (py-list* (py-list* 1 2 3) (py-list* 3 4 5))))
 (let ((b21
        (let ((e19
               (let ((e17 (get-global a)))
                 (let ((i16 0))
                   (cond
                    ((py-list? e17) (py-list-ref e17 i16))
                    ((tuple? e17) (tuple-ref e17 i16))
                    ((dict? e17) (dict-ref e17 i16))
                    (else (error "cannot index object")))))))
          (let ((i18 0))
            (cond
             ((py-list? e19) (py-list-ref e19 i18))
             ((tuple? e19) (tuple-ref e19 i18))
             ((dict? e19) (dict-ref e19 i18))
             (else (error "cannot index object")))))))
   (let ((i20 0))
     (cond
      ((tuple? b21) (tuple-set! b21 i20 2))
      ((py-list? b21) (py-list-set! b21 i20 2))
      ((dict? b21) (dict-set! b21 i20 2)))))
 (py-print
  (let ((e27
         (let ((e25
                (let ((e23 (get-global a)))
                  (let ((i22 0))
                    (cond
                     ((py-list? e23) (py-list-ref e23 i22))
                     ((tuple? e23) (tuple-ref e23 i22))
                     ((dict? e23) (dict-ref e23 i22))
                     (else (error "cannot index object")))))))
           (let ((i24 0))
             (cond
              ((py-list? e25) (py-list-ref e25 i24))
              ((tuple? e25) (tuple-ref e25 i24))
              ((dict? e25) (dict-ref e25 i24))
              (else (error "cannot index object")))))))
    (let ((i26 0))
      (cond
       ((py-list? e27) (py-list-ref e27 i26))
       ((tuple? e27) (tuple-ref e27 i26))
       ((dict? e27) (dict-ref e27 i26))
       (else (error "cannot index object")))))))
